packages/@aws-cdk/aws-ecs/lib/external/external-task-definition.ts

```diff
@@ -0,0 +1,83 @@
+import { Construct } from 'constructs';
+import { ImportedTaskDefinition } from '../../lib/base/_imported-task-definition';
+import {
+  CommonTaskDefinitionAttributes,
+  CommonTaskDefinitionProps,
+  Compatibility,
+  ITaskDefinition,
+  NetworkMode,
+  TaskDefinition,
+  Volume,
+} from '../base/task-definition';
+
+/**
+ * The properties for a task definition run on an External cluster.
+ */
+export interface ExternalTaskDefinitionProps extends CommonTaskDefinitionProps {
+
+}
+
+/**
+ * The interface of a task definition run on an External cluster.
+ */
+export interface IExternalTaskDefinition extends ITaskDefinition {
+
+}
+
+/**
+ * Attributes used to import an existing External task definition
+ */
+export interface ExternalTaskDefinitionAttributes extends CommonTaskDefinitionAttributes {
+
+}
+
+/**
+ * The details of a task definition run on an External cluster.
+ *
+ * @resource AWS::ECS::TaskDefinition
+ */
+export class ExternalTaskDefinition extends TaskDefinition implements IExternalTaskDefinition {
+
+  /**
+   * Imports a task definition from the specified task definition ARN.
+   */
+  public static fromEc2TaskDefinitionArn(scope: Construct, id: string, externalTaskDefinitionArn: string): IExternalTaskDefinition {
+    return new ImportedTaskDefinition(scope, id, {
+      taskDefinitionArn: externalTaskDefinitionArn,
+    });
+  }
+
+  /**
+   * Imports an existing External task definition from its attributes
+   */
+  public static fromExternalTaskDefinitionAttributes(
+    scope: Construct,
+    id: string,
+    attrs: ExternalTaskDefinitionAttributes,
+  ): IExternalTaskDefinition {
+    return new ImportedTaskDefinition(scope, id, {
+      taskDefinitionArn: attrs.taskDefinitionArn,
+      compatibility: Compatibility.EXTERNAL,
+      networkMode: NetworkMode.BRIDGE,
+      taskRole: attrs.taskRole,
+    });
+  }
+
+  /**
+   * Constructs a new instance of the ExternalTaskDefinition class.
+   */
+  constructor(scope: Construct, id: string, props: ExternalTaskDefinitionProps = {}) {
```

Comment:
We should validate that `networkMode !== awsvpc` in the constructor

Reply:
They can't change it, but what if it's set improperly at construction?